### DISCUSSION TITLE
Allow encryption between Proton and Simplelogin

### DISCUSSION THREAD

**Discussion by ghost** [simple-login/app#1416] [Feature Requests]
Hello. I suggest to encrypt everything on your side and transfer to proton inbox in encrypted form

**Comment by 4b15ed24f576** [comment]
Hey, you can actually do that already by adding your ProtonMail public PGP key to your mailbox. Although, keep in mind that this is a Premium feature.

Here is **how to add your public PGP key to SimpleLogin**:

1. Copy your public [Proton's PGP key](https://account.proton.me/u/2/mail/encryption-keys)
2. Go to [Mailboxes](https://app.simplelogin.io/dashboard/mailbox)
3. Add a new mailbox with your ProtonMail email address
4. Enable PGP and paste your public key

You're all set. From now on, any emails forwarded by your SimpleLogin alias to this mailbox will be encrypted with your ProtonMail public key.

**Comment by mpas97** [comment]
@acasajus I used the pgp encryption feature for many years. Now it has been disabled for Proton users ([commit](https://github.com/simple-login/app/commit/015036b499a66a39abee44c7b83f19515efcbbc2)). Why? I always knew that Proton encrypts everything that arrives unencrypted to store securely, but I had it enabled to secure the forwarding from simplelogin server to one of Proton. It has to be encrypted anyway, either here or one step later at Proton. So why are we no longer able to do that at the start of the secure chain?

**Comment by acasajus** [comment]
Emails from SL that have as destination a proton mailbox are sent directly from SL to proton without any intermediate step. We've disabled the PGP encryption in SL in this case to minimize the cases where there can be an error or where can users can make a mistake. For instance, we've had several cases where users uploaded the wrong key and ended up with unreadable emails. 
We understand that power users will be able to set it up flawlessly but even in that case there's no gain for that user but they add up maintenance they need to do. If they change the address keys they need to remember to update SL to keep it in sync and there's almost no security benefits from having the PGP encryption done in SL rather than in Proton.
